scipy/optimize/_linprog_rs.py

```diff
@@ -0,0 +1,540 @@
+    basis = np.concatenate((basis, basis_ng))
+    basis = _get_more_basis_columns(A, basis)  # add columns as needed
+
+    return A, b, c, basis, x, status
+
+
+def _select_singleton_columns(A, b):
+    """
+    Finds singleton columns for which the singleton entry is of the same sign
+    as the right hand side; these columns are eligible for inclusion in an
+    initial basis. Determines the rows in which the singleton entries are
+    located. For each of these rows, returns the indices of the one singleton
+    column and its corresponding row.
+    """
+    # find indices of all singleton columns and corresponding row indicies
+    column_indices = np.nonzero(np.sum(np.abs(A) != 0, axis=0) == 1)[0]
+    columns = A[:, column_indices]          # array of singleton columns
+    row_indices = np.zeros(len(column_indices), dtype=int)
+    nonzero_rows, nonzero_columns = np.nonzero(columns)
+    row_indices[nonzero_columns] = nonzero_rows   # corresponding row indicies
+
+    # keep only singletons with entries that have same sign as RHS
+    # this is necessary because all elements of BFS must be non-negative
+    same_sign = A[row_indices, column_indices]*b[row_indices] >= 0
+    column_indices = column_indices[same_sign][::-1]
+    row_indices = row_indices[same_sign][::-1]
+    # Reversing the order so that steps below select rightmost columns
+    # for initial basis, which will tend to be slack variables. (If the
+    # guess corresponds with a basic feasible solution but a constraint
+    # is not satisfied with the corresponding slack variable zero, the slack
+    # variable must be basic.)
+
+    # for each row, keep rightmost singleton column with an entry in that row
+    unique_row_indices, first_columns = np.unique(row_indices,
+                                                  return_index=True)
+    return column_indices[first_columns], unique_row_indices
+
+
+def _find_nonzero_rows(A, tol):
+    """
+    Returns logical array indicating the locations of rows with at least
+    one nonzero element.
+    """
+    return np.any(np.abs(A) > tol, axis=1)
+
+
+def _select_enter_pivot(c_hat, bl, a, rule="bland", tol=1e-12):
+    """
+    Selects a pivot to enter the basis. Currently Bland's rule - the smallest
+    index that has a negative reduced cost - is the default.
+    """
+    if rule.lower() == "mrc":  # index with minimum reduced cost
+        return a[~bl][np.argmin(c_hat)]
+    else:  # smallest index w/ negative reduced cost
+        return a[~bl][c_hat < -tol][0]
+
+
+def _display_iter(phase, iteration, slack, con, fun):
+    """
+    Print indicators of optimization status to the console.
+    """
+    header = True if not iteration % 20 else False
+
+    if header:
+        print("Phase",
+              "Iteration",
+              "Minimum Slack      ",
+              "Constraint Residual",
+              "Objective          ")
+
+    # no clue why this works
+    fmt = '{0:<6}{1:<10}{2:<20.13}{3:<20.13}{4:<20.13}'
```

Comment:
`<` means left-align, see https://www.python.org/dev/peps/pep-3101/#standard-format-specifiers. It's unusual, not sure if that's really what you want. Not sure if you want to use fixed point (`f`) or general (`g`) here?
```
In [1]: "{:20.13}".format(1.2342345239857320895409354803840)
Out[1]: '      1.234234523986'

In [2]: "{:20.13}".format(133333333.2342345239857320895409354803840)
Out[2]: '      133333333.2342'

In [3]: "{:1.6f}".format(1.2342345239857320895409354803840)
Out[3]: '1.234235'

In [4]: "{:1.6f}".format(133333333.2342345239857320895409354803840)
Out[4]: '133333333.234235'

In [5]: "{:1.6g}".format(1.2342345239857320895409354803840)
Out[5]: '1.23423'

In [6]: "{:1.6g}".format(133333333.2342345239857320895409354803840)
Out[6]: '1.33333e+08'

In [7]: "{:<20.13}".format(1.2342345239857320895409354803840)
Out[7]: '1.234234523986      '

In [8]: "{:<20.13}".format(133333333.2342345239857320895409354803840)
Out[8]: '133333333.2342      '
```

If you want to keep it, replace the "no clue" comment by "13 digits, left-aligned in 20 character spaces".

Reply:
I understood what it was supposed to do, but there was some funny business at the time. I seem to remember having to fiddle with it beyond what I thought I should have had to do.
Removed comment.
Should cleaning this up be added to #9671?